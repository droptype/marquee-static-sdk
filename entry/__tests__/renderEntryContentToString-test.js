const renderEntryContentToString = require('../renderEntryContentToString')

const SAMPLE_ENTRY = require('./SAMPLE_ENTRY.json')

describe('renderEntryContentToString', () => {
    it('renders', () => {
        const string = renderEntryContentToString(SAMPLE_ENTRY.content)
        expect(
            string
        ).toMatchSnapshot()
    })

    it('renders plain', () => {
        const string = renderEntryContentToString(SAMPLE_ENTRY.content, { plain: true })
        expect(
            string
        ).toMatchSnapshot()
    })

    it('renders empty', () => {
        const string = renderEntryContentToString([])
        expect(
            string
        ).toMatchSnapshot()
    })
})
