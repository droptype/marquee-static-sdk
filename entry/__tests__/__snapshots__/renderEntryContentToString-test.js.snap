// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderEntryContentToString renders 1`] = `"<p id=\\"text_7d112f487ec748c99aa022c7298c6965\\" class=\\"Block TextBlock -role--paragraph -align--left\\">Nullam quis <strong class=\\"Annotation -strong\\">risus</strong> eget urna mollis ornare vel eu leo. Donec sed odio dui. Sed posuere <em class=\\"Annotation -emphasis\\">consectetur</em> est at lobortis. Cum sociis natoque <strong class=\\"Annotation -strong\\">penatibus <em class=\\"Annotation -emphasis\\">et magnis</em></strong><em class=\\"Annotation -emphasis\\"> dis</em> parturient montes, nascetur ridiculus mus.</p><div id=\\"gallery_dfda4dd7fd5d445292f3d0d75ff462d3\\" class=\\"Block GalleryBlock -size--full\\" data-images=\\"[{&quot;urls&quot;:{&quot;128&quot;:&quot;https://s3.amazonaws.com/marquee-test-akiaisur2rgicbmpehea/RhXbcnHqRTis4rRYh7zi_casbah.jpg&quot;,&quot;640&quot;:&quot;https://s3.amazonaws.com/marquee-test-akiaisur2rgicbmpehea/XwQqlD5gQ8ONN7IhqDAj_casbah.jpg&quot;,&quot;1280&quot;:&quot;https://s3.amazonaws.com/marquee-test-akiaisur2rgicbmpehea/PjqA5y0KQ22sZTRt61xJ_casbah.jpg&quot;,&quot;2560&quot;:&quot;https://s3.amazonaws.com/marquee-test-akiaisur2rgicbmpehea/v7GsMybcST2bA9iqThym_casbah.jpg&quot;},&quot;credit&quot;:&quot;Me&quot;,&quot;caption&quot;:&quot;Oh hai&quot;,&quot;alt_text&quot;:&quot;A dog lies down.&quot;,&quot;aspect_ratio&quot;:1.3333333333333333},{&quot;urls&quot;:{&quot;128&quot;:&quot;https://s3.amazonaws.com/marquee-test-akiaisur2rgicbmpehea/KEDsGNwQdCfsTqqj25jx_3-original.jpg&quot;,&quot;640&quot;:&quot;https://s3.amazonaws.com/marquee-test-akiaisur2rgicbmpehea/Y9dGs9Z2RjalBC5s1y2l_3-original.jpg&quot;,&quot;1280&quot;:&quot;https://s3.amazonaws.com/marquee-test-akiaisur2rgicbmpehea/eYDtg85TuGWdAecAkWby_3-original.jpg&quot;,&quot;2560&quot;:&quot;https://s3.amazonaws.com/marquee-test-akiaisur2rgicbmpehea/2KCZptimSqyyDtzMSjMN_3-original.jpg&quot;},&quot;credit&quot;:&quot;&quot;,&quot;caption&quot;:&quot;caption&quot;,&quot;alt_text&quot;:&quot;&quot;,&quot;aspect_ratio&quot;:0.7499181669394436},{&quot;urls&quot;:{&quot;128&quot;:&quot;https://s3.amazonaws.com/marquee-test-akiaisur2rgicbmpehea/btz4fTLQLC9bafyfo8Ca_19-original.jpg&quot;,&quot;640&quot;:&quot;https://s3.amazonaws.com/marquee-test-akiaisur2rgicbmpehea/HEbcIUEWTYCziZ5yoMW7_19-original.jpg&quot;,&quot;1280&quot;:&quot;https://s3.amazonaws.com/marquee-test-akiaisur2rgicbmpehea/6MUTJ4hsTxuRhH6aDWa3_19-original.jpg&quot;,&quot;2560&quot;:&quot;https://s3.amazonaws.com/marquee-test-akiaisur2rgicbmpehea/dgaBWaZaTY2LJEDMv91I_19-original.jpg&quot;},&quot;aspect_ratio&quot;:1.5026178010471205},{&quot;urls&quot;:{&quot;128&quot;:&quot;https://s3.amazonaws.com/marquee-test-akiaisur2rgicbmpehea/c2OjmsXRKixvFlKQwKQN_32-original.jpg&quot;,&quot;640&quot;:&quot;https://s3.amazonaws.com/marquee-test-akiaisur2rgicbmpehea/Fe6jovYLQ52a2t8MJ9ZU_32-original.jpg&quot;,&quot;1280&quot;:&quot;https://s3.amazonaws.com/marquee-test-akiaisur2rgicbmpehea/AFDh62GuQgeJoUsqE22P_32-original.jpg&quot;,&quot;2560&quot;:&quot;https://s3.amazonaws.com/marquee-test-akiaisur2rgicbmpehea/4ilWlT9YQSSF3bBbxXCw_32-original.jpg&quot;},&quot;aspect_ratio&quot;:0.75},{&quot;urls&quot;:{&quot;128&quot;:&quot;https://s3.amazonaws.com/marquee-test-akiaisur2rgicbmpehea/tl1LEvQOQDWSucaQgezg_orly.jpg&quot;,&quot;640&quot;:&quot;https://s3.amazonaws.com/marquee-test-akiaisur2rgicbmpehea/PbliJcdSQ6u3gbr16AwY_orly.jpg&quot;,&quot;1280&quot;:&quot;https://s3.amazonaws.com/marquee-test-akiaisur2rgicbmpehea/3mEm0WoQPe41LaiBwbYB_orly.jpg&quot;,&quot;2560&quot;:&quot;https://s3.amazonaws.com/marquee-test-akiaisur2rgicbmpehea/D7YLrbixRjihwnybE9WH_orly.jpg&quot;},&quot;aspect_ratio&quot;:1.0755208333333333}]\\"><noscript><figure><img src=\\"https://s3.amazonaws.com/marquee-test-akiaisur2rgicbmpehea/XwQqlD5gQ8ONN7IhqDAj_casbah.jpg\\" alt=\\"A dog lies down.\\"/><figcaption><p>Oh hai</p><p>Me</p></figcaption></figure><figure><img src=\\"https://s3.amazonaws.com/marquee-test-akiaisur2rgicbmpehea/Y9dGs9Z2RjalBC5s1y2l_3-original.jpg\\" alt=\\"\\"/><figcaption><p>caption</p><p></p></figcaption></figure><figure><img src=\\"https://s3.amazonaws.com/marquee-test-akiaisur2rgicbmpehea/HEbcIUEWTYCziZ5yoMW7_19-original.jpg\\"/><figcaption><p></p><p></p></figcaption></figure><figure><img src=\\"https://s3.amazonaws.com/marquee-test-akiaisur2rgicbmpehea/Fe6jovYLQ52a2t8MJ9ZU_32-original.jpg\\"/><figcaption><p></p><p></p></figcaption></figure><figure><img src=\\"https://s3.amazonaws.com/marquee-test-akiaisur2rgicbmpehea/PbliJcdSQ6u3gbr16AwY_orly.jpg\\"/><figcaption><p></p><p></p></figcaption></figure></noscript></div><p id=\\"text_b457982144134dc8ba7fdcb9fac13300\\" class=\\"Block TextBlock -role--paragraph -align--left -effect--dropcaps\\">Aenean eu leo quam. Pellentesque ornare sem lacinia quam venenatis vestibulum. Fusce dapibus, tellus ac cursus commodo, tortor <a href=\\"http://example.com\\" data-external=\\"true\\" class=\\"Annotation -link\\">mauris</a> condimentum nibh, ut fermentum massa justo sit amet risus. Morbi leo risus, porta ac consectetur ac, vestibulum at eros. Fusce dapibus, tellus ac cursus commodo, tortor<sup class=\\"Annotation -superscript\\">2</sup> mauris condimentum<sub class=\\"Annotation -subscript\\">2</sub> nibh, ut <i class=\\"Annotation -small_caps\\">fermentum</i> massa justo sit amet risus. Integer posuere erat a ante venenatis dapibus posuere velit aliquet. Nullam id dolor id nibh ultricies vehicula ut id elit. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus.</p><ul class=\\"Block ListBlock -role--unordered\\" id=\\"list_efaa9d57ab614e6e88e9448a9cc7336e\\"><li>First <strong class=\\"Annotation -strong\\">item</strong></li><li>second</li><li>third?</li><li>fourth!</li></ul><figure id=\\"image_fe7b8a48f94b4ccb95a38beb91d00ff7\\" class=\\"Block ImageBlock -size--medium -position--right\\"><div class=\\"_Content\\"><img src=\\"https://s3.amazonaws.com/marquee-test-akiaisur2rgicbmpehea/Z4y6OgMSyK0fVGNQNcSn_DSC_6110.jpg\\" srcset=\\"https://s3.amazonaws.com/marquee-test-akiaisur2rgicbmpehea/saO09kR7TO6Fdkmpdo60_DSC_6110.jpg 128w,https://s3.amazonaws.com/marquee-test-akiaisur2rgicbmpehea/Z4y6OgMSyK0fVGNQNcSn_DSC_6110.jpg 640w,https://s3.amazonaws.com/marquee-test-akiaisur2rgicbmpehea/GLYe0WxSlem5uDSNxJ4M_DSC_6110.jpg 1280w,https://s3.amazonaws.com/marquee-test-akiaisur2rgicbmpehea/jKQgkPtRgSiIUoKbdoIc_DSC_6110.jpg 2560w\\" sizes=\\"(min-width: 720px) 720px,100vw\\" alt=\\"The spire of the Crysler building in the background, the tops of various skyscrapers in front of it.\\" data-src_128=\\"https://s3.amazonaws.com/marquee-test-akiaisur2rgicbmpehea/saO09kR7TO6Fdkmpdo60_DSC_6110.jpg\\" data-src_640=\\"https://s3.amazonaws.com/marquee-test-akiaisur2rgicbmpehea/Z4y6OgMSyK0fVGNQNcSn_DSC_6110.jpg\\" data-src_1280=\\"https://s3.amazonaws.com/marquee-test-akiaisur2rgicbmpehea/GLYe0WxSlem5uDSNxJ4M_DSC_6110.jpg\\" data-src_2560=\\"https://s3.amazonaws.com/marquee-test-akiaisur2rgicbmpehea/jKQgkPtRgSiIUoKbdoIc_DSC_6110.jpg\\"/><figcaption class=\\"_Caption\\"><p class=\\"_CaptionText\\">The Crysler building.</p><p class=\\"_Credit\\">Alec Perkins</p></figcaption></div></figure><p id=\\"text_62fdba4ae6374c00971e3d1c20a35ae0\\" class=\\"Block TextBlock -role--paragraph -align--left\\">Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Donec id elit non mi porta gravida at eget metus. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Duis mollis, est non commodo luctus, nisi erat porttitor ligula, eget lacinia odio sem nec elit. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus.</p><ol class=\\"Block ListBlock -role--ordered\\" id=\\"list_d00ef341f3c54e9a94cd76750cd81e0b\\"><li>A</li><li>B</li><li><em class=\\"Annotation -emphasis\\">C</em></li><li><sup class=\\"Annotation -superscript\\">D</sup></li><li>E</li><li>F</li><li>G</li></ol><figure class=\\"Block EmbedBlock -size--medium -position--center -source--embedly_html\\" id=\\"embed_47506d9d71e94c1b864d0f8b3b332b0b\\"><div class=\\"_Content\\"><div><blockquote class=\\"twitter-tweet\\"><a href=\\"https://twitter.com/marquee/status/832694046069960704\\"></a></blockquote><script async src=\\"//platform.twitter.com/widgets.js\\" charset=\\"utf-8\\"></script></div><figcaption class=\\"_Caption\\"><p class=\\"_CaptionText\\"></p><p class=\\"_Credit\\"></p></figcaption></div></figure><p id=\\"text_48063095ee214ca898cff298b42d7c29\\" class=\\"Block TextBlock -role--paragraph -align--left\\">Cras mattis consectetur purus sit amet fermentum. Maecenas faucibus mollis interdum. Curabitur blandit tempus porttitor. Cras mattis consectetur purus sit amet fermentum. Maecenas sed diam eget risus varius blandit sit amet non magna. Maecenas faucibus mollis interdum. Nullam quis risus eget urna mollis ornare vel eu leo.</p>"`;

exports[`renderEntryContentToString renders empty 1`] = `""`;

exports[`renderEntryContentToString renders plain 1`] = `"<p>Nullam quis <strong>risus</strong> eget urna mollis ornare vel eu leo. Donec sed odio dui. Sed posuere <em>consectetur</em> est at lobortis. Cum sociis natoque <strong>penatibus <em>et magnis</em></strong><em> dis</em> parturient montes, nascetur ridiculus mus.</p><div type=\\"figure\\"><figure><img src=\\"https://s3.amazonaws.com/marquee-test-akiaisur2rgicbmpehea/Y9dGs9Z2RjalBC5s1y2l_3-original.jpg\\" alt=\\"\\"/><figcaption><p>caption</p><p></p></figcaption></figure><figure><img src=\\"https://s3.amazonaws.com/marquee-test-akiaisur2rgicbmpehea/HEbcIUEWTYCziZ5yoMW7_19-original.jpg\\"/><figcaption><p></p><p></p></figcaption></figure><figure><img src=\\"https://s3.amazonaws.com/marquee-test-akiaisur2rgicbmpehea/Fe6jovYLQ52a2t8MJ9ZU_32-original.jpg\\"/><figcaption><p></p><p></p></figcaption></figure><figure><img src=\\"https://s3.amazonaws.com/marquee-test-akiaisur2rgicbmpehea/PbliJcdSQ6u3gbr16AwY_orly.jpg\\"/><figcaption><p></p><p></p></figcaption></figure></div><p>Aenean eu leo quam. Pellentesque ornare sem lacinia quam venenatis vestibulum. Fusce dapibus, tellus ac cursus commodo, tortor <a href=\\"http://example.com\\">mauris</a> condimentum nibh, ut fermentum massa justo sit amet risus. Morbi leo risus, porta ac consectetur ac, vestibulum at eros. Fusce dapibus, tellus ac cursus commodo, tortor<sup>2</sup> mauris condimentum<sub>2</sub> nibh, ut fermentum massa justo sit amet risus. Integer posuere erat a ante venenatis dapibus posuere velit aliquet. Nullam id dolor id nibh ultricies vehicula ut id elit. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus.</p><ul><li>First <strong>item</strong></li><li>second</li><li>third?</li><li>fourth!</li></ul><figure><img src=\\"https://s3.amazonaws.com/marquee-test-akiaisur2rgicbmpehea/Z4y6OgMSyK0fVGNQNcSn_DSC_6110.jpg\\"/><figcaption><p>The Crysler building.</p><p>Alec Perkins</p></figcaption></figure><p>Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Donec id elit non mi porta gravida at eget metus. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Duis mollis, est non commodo luctus, nisi erat porttitor ligula, eget lacinia odio sem nec elit. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus.</p><ol><li>A</li><li>B</li><li><em>C</em></li><li><sup>D</sup></li><li>E</li><li>F</li><li>G</li></ol><figure><div><blockquote class=\\"twitter-tweet\\"><a href=\\"https://twitter.com/marquee/status/832694046069960704\\"></a></blockquote><script async src=\\"//platform.twitter.com/widgets.js\\" charset=\\"utf-8\\"></script></div><figcaption><p></p><p></p></figcaption></figure><p>Cras mattis consectetur purus sit amet fermentum. Maecenas faucibus mollis interdum. Curabitur blandit tempus porttitor. Cras mattis consectetur purus sit amet fermentum. Maecenas sed diam eget risus varius blandit sit amet non magna. Maecenas faucibus mollis interdum. Nullam quis risus eget urna mollis ornare vel eu leo.</p>"`;
